### PR TITLE
Tools: Call-Tree Tool

### PR DESCRIPTION
**Pull request by stefanodellosa**
Method to run preprocessor on source files added.

TODOs:
- Fix handing of function pointers

Signed-off-by: Stefano Dell'Osa <stefano.dellosa@intel.com>

### CODE REVIEW

File: call_tree_tool/edgeprocessor.py [hunk 1]
```diff
@@ -50,7 +50,13 @@ def process_line(self, line):
 
         funcrecord = self.database.get_function_by_name(funcname)
         if not funcrecord:
-            funcdata = ("unknown", funcname)
+            # check if function is in filter list (E.g. Compiler extensions to C language)
+            with open("func_filter", "r") as filtered_funcs:
```
Comment (eligurvitz) [inline call_tree_tool/edgeprocessor.py:54]: You mention the file func_filter - is it a new file that we add to the release? Or does the user have to provide this file? In both cases, more information is needed.

**Review by eligurvitz — CHANGES_REQUESTED** [review]
Looks good. I added one comment regarding the presumably new file "func_filter"

**Comment by eligurvitz** [comment]
Changes look good. I submitted one comment regarding the presumably new file: "func_filter".

**Review by paolonig — COMMENTED** [review]
- TODOs must be embedded in the code as comments, otherwise they'll get lost. So it is ok to mention them in the commit msg but it is important to add them to the code too.
- The commit msg must be more verbose and explanatory of the change. I.e.

call_tree_tool: [summary phrase]

                [verbose description of what is the problem fix or the feature added]

Signed-off-by: [your-name and email]

File: call_tree_tool/edgeprocessor.py (hunk 1 above)
Comment (stefanodellosa) [reply call_tree_tool/edgeprocessor.py:54]: Added info into Readme file

**Comment by stefanodellosa** [comment]
> * TODOs must be embedded in the code as comments, otherwise they'll get lost. So it is ok to mention them in the commit msg but it is important to add them to the code too.
> * The commit msg must be more verbose and explanatory of the change. I.e.
> 
> call_tree_tool: [summary phrase]
> 
> ```
>             [verbose description of what is the problem fix or the feature added]
> ```
> 
> Signed-off-by: [your-name and email]

Done.

**Review by paolonig — APPROVED** [review]
Looks good to me

**Review by paolonig — APPROVED** [review]

**Pull request merged**